notebooks/search/07-inference.ipynb

```diff
@@ -0,0 +1,382 @@
+    "        \"type\": \"dense_vector\", \n",
+    "        \"dims\": 1536,\n",
+    "        \"similarity\": \"dot_product\" \n",
+    "      },\n",
+    "      \"plot\": { \n",
+    "        \"type\": \"text\" \n",
+    "      }\n",
+    "    }\n",
+    "  }\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e9d4bfd2",
+   "metadata": {},
+   "source": [
+    "## Insert Documents\n",
+    "\n",
+    "Let's insert our example dataset of 12 movies. You need a paid OpenAI account to complete this step, otherwise the documentation ingest will time out due to the API request rate limits."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "cfa8eda5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "url = \"https://raw.githubusercontent.com/elastic/elasticsearch-labs/main/notebooks/search/movies.json\"\n",
+    "response = urlopen(url)\n",
+    "\n",
+    "# Load the response data into a JSON object\n",
+    "data_json = json.loads(response.read())\n",
+    "\n",
+    "# Prepare the documents to be indexed\n",
+    "documents = []\n",
+    "for doc in data_json:\n",
+    "    documents.append({\n",
+    "        \"_index\": \"openai-movie-embeddings\",\n",
+    "        \"_source\": doc,\n",
+    "    })\n",
+    "\n",
+    "# Use helpers.bulk to index\n",
+    "helpers.bulk(client, documents)\n",
+    "\n",
+    "print(\"Done indexing documents into `openai-movie-embeddings` index!\")\n",
+    "time.sleep(3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a68e808e",
+   "metadata": {},
+   "source": [
+    "## Semantic search\n",
+    "\n",
+    "After the dataset has been enriched with the embeddings, you can query the data using [semantic search](https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#knn-semantic-search). Pass a `query_vector_builder` to the k-nearest neighbor (kNN) vector search API, and provide the query text and the model you have used to create the embeddings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a47cdc60",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "response = client.search(\n",
+    "    index='openai-movie-embeddings', \n",
+    "    size=3,\n",
+    "    knn={\n",
+    "        \"field\": \"plot_embedding\",\n",
+    "        \"query_vector_builder\": {\n",
+    "            \"text_embedding\": {\n",
+    "                \"model_id\": \"openai_embeddings\",\n",
+    "                \"model_text\": \"Fighting movie\"\n",
+    "            }\n",
+    "        },\n",
+    "        \"k\": 10,\n",
+    "        \"num_candidates\": 100\n",
+    "        }\n",
+    ")\n",
+    "\n",
+    "for hit in response['hits']['hits']:\n",
+    "    doc_id = hit['_id']\n",
+    "    score = hit['_score']\n",
+    "    title = hit['_source']['title']\n",
+    "    plot = hit['_source']['plot']\n",
+    "    print(f\"Score: {score}\\nTitle: {title}\\nPlot: {plot}\\n\")"
```

Comment:
I think it would be nice if you save your notebook with the output from the search, so that people who are viewing it without being all set up to run it can see what the results are too.